### PR TITLE
VZ-2347: More changes to support authorization in the VMI stack

### PR DESCRIPTION
**Pull request by fredtibbitts**
# Description

This PR includes the following:

- Pulls in the latest version of the verrazzano-monitoring-operator with support for VMI authorization and Grafana identity propagation
- Updates the Keycloak configuration to make the verrazzano-oauth-client public, adds console_user as a default realm role for users, and adds a mapper to the webui and verrazzano-oauth-client clients so that realm roles are included in the id_token
- Fixes a Lua script bug with user impersonation in the API proxy.

Implements VZ-2347

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/scripts/install/config/keycloak.json [hunk 1]
```diff
@@ -777,6 +759,21 @@
...
       }
     }, {
+      "id": "67ef034c-ab84-42d3-9428-b2d6ca4d8979",
+      "name": "realm roles",
+      "protocol": "openid-connect",
+      "protocolMapper": "oidc-usermodel-realm-role-mapper",
+      "consentRequired": false,
+      "config": {
+        "multivalued": "true",
+        "user.attribute": "foo",
```
Comment (wmhopkins) [inline platform-operator/scripts/install/config/keycloak.json:769]: What are we using the "foo" attribute for? ;)

File: platform-operator/scripts/install/config/keycloak.json [hunk 2]
```diff
@@ -905,7 +902,22 @@
...
+        },
+        {
+          "id": "5a713e4a-b114-4780-9eb8-2e3aa43e20b2",
+          "name": "realm roles",
+          "protocol": "openid-connect",
+          "protocolMapper": "oidc-usermodel-realm-role-mapper",
+          "consentRequired": false,
+          "config": {
+            "multivalued": "true",
+            "user.attribute": "foo",
```
Comment (wmhopkins) [inline platform-operator/scripts/install/config/keycloak.json:914]: Also here (foo).

**Review by wmhopkins — COMMENTED** [review]
Looks good, but do we need to remove the "foo" attribute?

File: platform-operator/scripts/install/config/keycloak.json (hunk 1 above)
Comment (fredtibbitts) [reply platform-operator/scripts/install/config/keycloak.json:769]: Keycloak generates that when you export the configuration (it appears in other places in this file as well). I wasn't confident enough to know that deleting that wouldn't have negative consequences so I just left it in.

**Review by wmhopkins — APPROVED** [review]

**Pull request merged**